src/components/tree-views-container.tsx

```diff
@@ -360,15 +360,20 @@ const TreeViewsContainer = () => {
                         [[] as UploadingElement[], [] as UploadingElement[]]
                     );
 
-                // then remove the ghosts if necessary
+                // then remove the ghosts if the upload succeeded
                 if (toRemoveFromUploadingElements.length > 0) {
                     let newUploadingElements = { ...uploadingElementsRef.current };
-                    toRemoveFromUploadingElements.forEach((r) => delete newUploadingElements[r.id]);
-
+                    if (toRemoveFromUploadingElements.length > 0) {
+                        toRemoveFromUploadingElements.forEach((r) => delete newUploadingElements[r.id]);
+                    }
                     dispatch(setUploadingElements(newUploadingElements));
                 }
-
-                return [...current, ...toKeepToUploadingElements].sort(function (a, b) {
+                //remove the ghosts if the upload failed
+                let newCurrentElement = cleanCreationFailedElementInCurrentElements(
+                    current,
+                    uploadingElementsInSelectedDirectory
+                );
+                return [...newCurrentElement, ...toKeepToUploadingElements].sort(function (a, b) {
                     return a.elementName.localeCompare(b.elementName);
                 }) as ElementAttributes[];
             } else {
```

Comment:
We need to call `cleanCreationFailedElementInCurrentElements` in the else case as well because if `uploadingElementsInSelectedDirectory` is empty `current` could still have a ghost element to clean